waku/incentivization/txid_proof.nim

```diff
@@ -0,0 +1,62 @@
+import std/options, chronos, web3, stew/byteutils, stint, strutils
+
+import waku/incentivization/rpc
+
+
+# Function to convert a hex string to Address
+proc toAddress*(hexStr: string): Address =
+  # Remove the "0x" prefix if it exists
+  let cleaned = if hexStr.startsWith("0x"): hexStr[2..^1] else: hexStr
+
+  # Ensure the length is exactly 40 characters (20 bytes)
+  if cleaned.len != 40:
+    raise newException(ValueError, "Invalid hexadecimal string length for Address")
+
+  var arr: array[20, byte]
+  for i in 0 ..< 20:
+    let byteValue = cleaned[i * 2 ..< i * 2 + 2]  # Get two hex characters
+    arr[i] = byte(parseHexInt(byteValue))
+
+  result = Address(arr)
+
+
+proc checkTxIdIsEligible(txHash: TxHash, ethClient: string): Future[bool] {.async.} =
+  let web3 = await newWeb3(ethClient)
+  try:
```

Comment:
`try...except` should preferably just be around the block of code that can actually raise the exception you want to catch.

Reply:
What would be the preferred way to use the value defined within a `try`-block? In this code, only querying the RPC can raise the exception, so ideally the `try`-block would only contain the two queries. But when I assign, for example, `let tx = await web3.provider.eth_getTransactionByHash(txHash)`, `tx` remains in scope only within the `try`-block. I could define `tx` before the `try`-block and then assign the value to it, but I'm not sure it's a good idea either to initialize a variable with a `none` value... (It should be an Option type then?)